makelib/golang.mk

```diff
@@ -277,22 +306,22 @@ fmt: go.imports
 fmt.simplify: go.fmt.simplify
 imports: go.imports
 imports.fix: go.imports.fix
-vendor: go.vendor
+vendor: $(GO_VENDOR_TARGET)
 vendor.check: go.vendor.check
 vendor.update: go.vendor.update
 vet: go.vet
-generate codegen: go.generate
 
 define GO_HELPTEXT
 Go Targets:
-    generate        Runs go code generation followed by goimports on generated files.
     fmt             Checks go source code for formatting issues.
     fmt.simplify    Format, simplify, update source files.
     imports         Checks go source code import lines for issues.
     imports.fix     Updates go source files to fix issues with import lines.
     vendor          Updates vendor packages.
     vendor.check    Fail the build if vendor packages have changed.
     vendor.update   Update vendor dependencies.
+    go.mod.verify      Verify go module dependencies have expected content.
```

Comment:
This is the reason why I created a separate file for go modules in the first iteration. 

`vendor` folder is not relevant in `go modules`, as it is in `dep`.  Even though there is a `go mod vendor` command to populate the `vendor` folder, but the default source of truth for go packages is  `~/go/pkg/mod`

Because of this, I'm not sure if a `vendor` target for go modules is really meaningful

Reply:
removed the new targets, in favor of existing vendor targets, keeping the semantics